source/iterator/cdc.go

```diff
@@ -220,23 +235,46 @@ func (c *CDC) routeItem(
 		return fmt.Errorf("marshal sdk position: %w", err)
 	}
 
+	c.records <- c.GetRecord(item,
+		itemCreatedAt,
+		itemUpdatedAt,
+		itemDeletedAt,
+		updatedAfter,
+		sdkPosition,
+		metadata)
+
+	return nil
+}
+
+func (c *CDC) GetRecord(item hubspot.ListResponseResult,
+	itemCreatedAt,
+	itemUpdatedAt,
+	itemDeletedAt,
+	updatedAfter time.Time,
+	sdkPosition sdk.Position,
+	metadata sdk.Metadata,
+) sdk.Record {
+	if itemDeletedAt != (time.Time{}) {
+		if itemDeletedAt == itemUpdatedAt && itemDeletedAt.Unix() > 0 {
+			return sdk.Util.Source.NewRecordDelete(sdkPosition, metadata,
+				sdk.StructuredData{hubspot.ResultsFieldID: item[hubspot.ResultsFieldID]},
+			)
+		}
+	}
```

Comment:
Hm, after looking at this for a little bit longer, I realized that the following would be enough:
```go
if itemDeletedAt.Unix() > 0 {
	return sdk.Util.Source.NewRecordDelete(sdkPosition, metadata,
		sdk.StructuredData{hubspot.ResultsFieldID: item[hubspot.ResultsFieldID]},
	)
}
```